From e0e3aca997f7f2b9b5b5e27f878e9bf5b573720a Mon Sep 17 00:00:00 2001
From: Stephen Boyd <sboyd@kernel.org>
Date: Tue, 4 Apr 2023 13:45:53 -0700
Subject: clk: mediatek: Use right match table, include mod_devicetable

This is copy/pasta that breaks modular builds. Fix the match table to
use the right pointer, or the right device table type. And while we're
including the header, fix the order to be linux, dt-bindings, and
finally local.

Cc: Garmin.Chang <Garmin.Chang@mediatek.com>
Cc: Chen-Yu Tsai <wenst@chromium.org>
Reviewed-by: AngeloGioacchino Del Regno <angelogioacchino.delregno@collabora.com>
Fixes: f42b9e9a43e3 ("clk: mediatek: Add MT8188 wpesys clock support")
Fixes: 0d2f2cefba64 ("clk: mediatek: Add MT8188 adsp clock support")
Fixes: e4aaa60eae16 ("clk: mediatek: Add MT8188 vdosys0 clock support")
Fixes: cfa4609f9bbe ("clk: mediatek: Add MT8188 vdosys1 clock support")
Fixes: bb87c1109ce2 ("clk: mediatek: Add MT8188 vencsys clock support")
Reported-by: kernel test robot <lkp@intel.com>
Link: https://lore.kernel.org/oe-kbuild-all/202304011039.UBDX1UOT-lkp@intel.com/
Link: https://lore.kernel.org/oe-kbuild-all/202304020649.QO2HlpD5-lkp@intel.com/
Link: https://lore.kernel.org/oe-kbuild-all/202304021055.WDhQPcoS-lkp@intel.com/
Signed-off-by: Stephen Boyd <sboyd@kernel.org>
Link: https://lore.kernel.org/r/20230404204553.1256263-1-sboyd@kernel.org
---
 drivers/clk/mediatek/clk-mt8188-adsp_audio26m.c | 6 ++++--
 drivers/clk/mediatek/clk-mt8188-imp_iic_wrap.c  | 6 ++++--
 drivers/clk/mediatek/clk-mt8188-vdo0.c          | 4 +++-
 drivers/clk/mediatek/clk-mt8188-vdo1.c          | 4 +++-
 drivers/clk/mediatek/clk-mt8188-venc.c          | 6 ++++--
 drivers/clk/mediatek/clk-mt8188-wpe.c           | 6 ++++--
 6 files changed, 22 insertions(+), 10 deletions(-)

diff --git a/drivers/clk/mediatek/clk-mt8188-adsp_audio26m.c b/drivers/clk/mediatek/clk-mt8188-adsp_audio26m.c
index b3c31ffff9311..808f2ad3b7ee6 100644
--- a/drivers/clk/mediatek/clk-mt8188-adsp_audio26m.c
+++ b/drivers/clk/mediatek/clk-mt8188-adsp_audio26m.c
@@ -4,10 +4,12 @@
  * Author: Garmin Chang <garmin.chang@mediatek.com>
  */
 
-#include <dt-bindings/clock/mediatek,mt8188-clk.h>
 #include <linux/clk-provider.h>
+#include <linux/mod_devicetable.h>
 #include <linux/platform_device.h>
 
+#include <dt-bindings/clock/mediatek,mt8188-clk.h>
+
 #include "clk-gate.h"
 #include "clk-mtk.h"
 
@@ -34,7 +36,7 @@ static const struct of_device_id of_match_clk_mt8188_adsp_audio26m[] = {
 	{ .compatible = "mediatek,mt8188-adsp-audio26m", .data = &adsp_audio26m_desc },
 	{ /* sentinel */ }
 };
-MODULE_DEVICE_TABLE(platform, of_match_clk_mt8188_adsp_audio26m);
+MODULE_DEVICE_TABLE(of, of_match_clk_mt8188_adsp_audio26m);
 
 static struct platform_driver clk_mt8188_adsp_audio26m_drv = {
 	.probe = mtk_clk_simple_probe,
diff --git a/drivers/clk/mediatek/clk-mt8188-imp_iic_wrap.c b/drivers/clk/mediatek/clk-mt8188-imp_iic_wrap.c
index 2238d29843200..da41a3c59919c 100644
--- a/drivers/clk/mediatek/clk-mt8188-imp_iic_wrap.c
+++ b/drivers/clk/mediatek/clk-mt8188-imp_iic_wrap.c
@@ -4,10 +4,12 @@
  * Author: Garmin Chang <garmin.chang@mediatek.com>
  */
 
-#include <dt-bindings/clock/mediatek,mt8188-clk.h>
 #include <linux/clk-provider.h>
+#include <linux/mod_devicetable.h>
 #include <linux/platform_device.h>
 
+#include <dt-bindings/clock/mediatek,mt8188-clk.h>
+
 #include "clk-gate.h"
 #include "clk-mtk.h"
 
@@ -65,7 +67,7 @@ static const struct of_device_id of_match_clk_mt8188_imp_iic_wrap[] = {
 	{ .compatible = "mediatek,mt8188-imp-iic-wrap-en", .data = &imp_iic_wrap_en_desc },
 	{ /* sentinel */ }
 };
-MODULE_DEVICE_TABLE(platform, of_match_clk_mt8188_imp_iic_wrap);
+MODULE_DEVICE_TABLE(of, of_match_clk_mt8188_imp_iic_wrap);
 
 static struct platform_driver clk_mt8188_imp_iic_wrap_drv = {
 	.probe = mtk_clk_simple_probe,
diff --git a/drivers/clk/mediatek/clk-mt8188-vdo0.c b/drivers/clk/mediatek/clk-mt8188-vdo0.c
index 0c61c2dc43375..d2be44c2f3f58 100644
--- a/drivers/clk/mediatek/clk-mt8188-vdo0.c
+++ b/drivers/clk/mediatek/clk-mt8188-vdo0.c
@@ -4,10 +4,12 @@
  * Author: Garmin Chang <garmin.chang@mediatek.com>
  */
 
-#include <dt-bindings/clock/mediatek,mt8188-clk.h>
 #include <linux/clk-provider.h>
+#include <linux/mod_devicetable.h>
 #include <linux/platform_device.h>
 
+#include <dt-bindings/clock/mediatek,mt8188-clk.h>
+
 #include "clk-gate.h"
 #include "clk-mtk.h"
 
diff --git a/drivers/clk/mediatek/clk-mt8188-vdo1.c b/drivers/clk/mediatek/clk-mt8188-vdo1.c
index 99fcf6d7b1abf..2ef8cae2e16ed 100644
--- a/drivers/clk/mediatek/clk-mt8188-vdo1.c
+++ b/drivers/clk/mediatek/clk-mt8188-vdo1.c
@@ -4,10 +4,12 @@
  * Author: Garmin Chang <garmin.chang@mediatek.com>
  */
 
-#include <dt-bindings/clock/mediatek,mt8188-clk.h>
 #include <linux/clk-provider.h>
+#include <linux/mod_devicetable.h>
 #include <linux/platform_device.h>
 
+#include <dt-bindings/clock/mediatek,mt8188-clk.h>
+
 #include "clk-gate.h"
 #include "clk-mtk.h"
 
diff --git a/drivers/clk/mediatek/clk-mt8188-venc.c b/drivers/clk/mediatek/clk-mt8188-venc.c
index 6f6589ccd5a01..245367f33fa53 100644
--- a/drivers/clk/mediatek/clk-mt8188-venc.c
+++ b/drivers/clk/mediatek/clk-mt8188-venc.c
@@ -4,10 +4,12 @@
  * Author: Garmin Chang <garmin.chang@mediatek.com>
  */
 
-#include <dt-bindings/clock/mediatek,mt8188-clk.h>
 #include <linux/clk-provider.h>
+#include <linux/mod_devicetable.h>
 #include <linux/platform_device.h>
 
+#include <dt-bindings/clock/mediatek,mt8188-clk.h>
+
 #include "clk-gate.h"
 #include "clk-mtk.h"
 
@@ -39,7 +41,7 @@ static const struct of_device_id of_match_clk_mt8188_venc1[] = {
 	{ .compatible = "mediatek,mt8188-vencsys", .data = &venc1_desc },
 	{ /* sentinel */ }
 };
-MODULE_DEVICE_TABLE(platform, of_match_clk_mt8188_venc1);
+MODULE_DEVICE_TABLE(of, of_match_clk_mt8188_venc1);
 
 static struct platform_driver clk_mt8188_venc1_drv = {
 	.probe = mtk_clk_simple_probe,
diff --git a/drivers/clk/mediatek/clk-mt8188-wpe.c b/drivers/clk/mediatek/clk-mt8188-wpe.c
index 5abded13cecec..393ac38a21724 100644
--- a/drivers/clk/mediatek/clk-mt8188-wpe.c
+++ b/drivers/clk/mediatek/clk-mt8188-wpe.c
@@ -4,10 +4,12 @@
  * Author: Garmin Chang <garmin.chang@mediatek.com>
  */
 
-#include <dt-bindings/clock/mediatek,mt8188-clk.h>
 #include <linux/clk-provider.h>
+#include <linux/mod_devicetable.h>
 #include <linux/platform_device.h>
 
+#include <dt-bindings/clock/mediatek,mt8188-clk.h>
+
 #include "clk-gate.h"
 #include "clk-mtk.h"
 
@@ -88,7 +90,7 @@ static const struct of_device_id of_match_clk_mt8188_wpe[] = {
 	{ .compatible = "mediatek,mt8188-wpesys-vpp0", .data = &wpe_vpp0_desc },
 	{ /* sentinel */ }
 };
-MODULE_DEVICE_TABLE(platform, clk_mt8188_vpp1_id_table);
+MODULE_DEVICE_TABLE(of, of_match_clk_mt8188_wpe);
 
 static struct platform_driver clk_mt8188_wpe_drv = {
 	.probe = mtk_clk_simple_probe,
-- 
cgit 

